### DISCUSSION TITLE
foundry and unraid docker issues

### DISCUSSION THREAD

**Discussion by Slothy2406** [felddy/foundryvtt-docker#787] [Q&A] [answered]
Hi all, 
I am hoping that someone might be able to help me.

I have managed to install the docker image on my unraid system (via the silman app) with no issues, but when I go to upgrade it I always run into issues, such as the system wipes all the game systems, world and addons from within Foundry. I don't know if I am missing something from the initial config. I also seem to only be able to install from the timed URL from the Foundryvtt website. When I just use my username and password, it seems to authenticate correctly but just wont download the latest release (unless I enter something in to the FOUNDARY_VERSION option, but I just want to leave that blank and it pull the latest version when an upgrade comes round and keep all the current foundry setup).

So if anyone is able to point me in the right direction that would be absolutely amazing. Please find below the logs and config for my image.

Authenticate | 2023-09-13 12:49:12 | [debug] Community URL: /community/xxx
Authenticate | 2023-09-13 12:49:12 | [info] Successfully logged in as: xxx
ReleaseURL | 2023-09-13 12:49:13 | [debug] Loading cookies from: cookiejar.json
ReleaseURL | 2023-09-13 12:49:13 | [error] Unable to extract build number from version: 
file:///home/foundry/get_release_url.js:96
        throw new Error(`Unable to extract build number from version: ${foundry_version}`);
              ^

Error: Unable to extract build number from version: 
    at main (file:///home/foundry/get_release_url.js:96:15)
    at file:///home/foundry/get_release_url.js:109:30
    at file:///home/foundry/get_release_url.js:110:3
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.17.1
Entrypoint | 2023-09-13 12:49:12 | [info] Using authenticated credentials to download release.
Entrypoint | 2023-09-13 12:50:18 | [debug] Timezone set to: UTC
Entrypoint | 2023-09-13 12:50:18 | [info] Starting felddy/foundryvtt container v11.308.0
Authenticate | 2023-09-13 12:50:19 | [debug] Saving cookies to: cookiejar.json
Authenticate | 2023-09-13 12:50:19 | [info] Requesting CSRF tokens from https://foundryvtt.com
Authenticate | 2023-09-13 12:50:19 | [debug] Fetching: https://foundryvtt.com
Authenticate | 2023-09-13 12:50:20 | [info] Logging in as: xxx
Entrypoint | 2023-09-13 12:50:18 | [debug] CONTAINER_VERBOSE set.  Debug logging enabled.
Authenticate | 2023-09-13 12:50:20 | [debug] Fetching: https://foundryvtt.com/auth/login/
Authenticate | 2023-09-13 12:50:22 | [debug] Community URL: /community/xxx
Authenticate | 2023-09-13 12:50:22 | [info] Successfully logged in as: xxx
ReleaseURL | 2023-09-13 12:50:22 | [debug] Loading cookies from: cookiejar.json
ReleaseURL | 2023-09-13 12:50:22 | [error] Unable to extract build number from version: 
file:///home/foundry/get_release_url.js:96
        throw new Error(`Unable to extract build number from version: ${foundry_version}`);
Entrypoint | 2023-09-13 12:50:18 | [debug] Running as: uid=0(root) gid=0(root) groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)              ^

Error: Unable to extract build number from version: 
    at main (file:///home/foundry/get_release_url.js:96:15)
    at file:///home/foundry/get_release_url.js:109:30
    at file:///home/foundry/get_release_url.js:110:3
Entrypoint | 2023-09-13 12:50:18 | [debug] Environment: CONTAINER_CACHE=    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.17.1
CONTAINER_PATCHES=
CONTAINER_PATCH_URLS=
CONTAINER_PRESERVE_CONFIG=true
CONTAINER_PRESERVE_OWNER=
CONTAINER_VERBOSE=false
FOUNDRY_ADMIN_KEY=[REDACTED]
FOUNDRY_AWS_CONFIG=
FOUNDRY_DEMO_CONFIG=
FOUNDRY_GID=foundry
FOUNDRY_HOME=/home/foundry
FOUNDRY_HOSTNAME=
FOUNDRY_IP_DISCOVERY=true
FOUNDRY_LANGUAGE=en.core
FOUNDRY_LICENSE_KEY=[REDACTED]
FOUNDRY_LOCAL_HOSTNAME=
FOUNDRY_MINIFY_STATIC_FILES=true
FOUNDRY_PASSWORD=[REDACTED]
FOUNDRY_PASSWORD_SALT=
FOUNDRY_PROXY_PORT=
FOUNDRY_PROXY_SSL=false
FOUNDRY_RELEASE_URL=
FOUNDRY_ROUTE_PREFIX=
FOUNDRY_SSL_CERT=
FOUNDRY_SSL_KEY=[REDACTED]
FOUNDRY_UID=foundry
FOUNDRY_UPNP=false
FOUNDRY_UPNP_LEASE_DURATION=
FOUNDRY_USERNAME=xxx
FOUNDRY_VERSION=
FOUNDRY_WORLD=
HOME=/root
HOSTNAME=6ff46d43cc18
HOST_CONTAINERNAME=Foundry
HOST_HOSTNAME=unraid
HOST_OS=Unraid
NODE_VERSION=18.17.1
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/home/foundry
SHLVL=1
TCP_PORT_30000=30000
TIMEZONE=UTC
TZ=Europe/London
YARN_VERSION=1.22.19
Entrypoint | 2023-09-13 12:50:18 | [warn] FOUNDRY_VERSION has been manually set and does not match the container's version.
Entrypoint | 2023-09-13 12:50:18 | [warn] Expected 11.308 but found 
Entrypoint | 2023-09-13 12:50:18 | [warn] The container may not function properly with this version mismatch.
Entrypoint | 2023-09-13 12:50:18 | [info] No Foundry Virtual Tabletop installation detected.
Entrypoint | 2023-09-13 12:50:18 | [info] Using FOUNDRY_USERNAME and FOUNDRY_PASSWORD to authenticate.
Entrypoint | 2023-09-13 12:50:22 | [info] Using authenticated credentials to download release.

**Comment by felddy** [comment, marked as answer]
You are setting the `FOUNDRY_VERSION` environment variable to "" (the empty string).  I'm not familiar with the interface you are using, but you need to delete the variable from your config (which is different from setting it to "").   If that is not possible, then set it to `11.308`, which is the default value for this version of the container.  

See:
- https://github.com/felddy/foundryvtt-docker#optional-variables

**Reply by Slothy2406** [reply]
Thanks for the quick reply.

And after reading up more about FVTT, then it is probably best to keep the version fairly static unless there is an issue, so putting in a value wont suddenly upgrade your server on a reboot.

Thanks Again
Andy
